### DISCUSSION TITLE
Typed Link Field added to Craft 5 Core Functionally

### DISCUSSION THREAD

**Discussion by kellyannmcnamara01** [craftcms/cms#14962] [Ideas] [answered]
A plugin called "[Typed Link Field](https://plugins.craftcms.com/typedlinkfield)" has been abandoned for Craft 5. As of today it still has ~32k installs. The plugin adds a link field that seems like it would be part of the core of Craft. Can it's actions be added to the core of craft, and can there be a conversation command?

The field allows a few settings: what type of links can be created, the link text and if it opens in a new tab of not. Depending on the type of link there is a forth field that appears in alignment. Below are the images of what the field looks like in the admin to better explain the request of this being part of the core.

Bonus request: It would be nice is CKeditor could adapt part of this requested core functionally too

<img width="1786" alt="image" src="https://github.com/craftcms/cms/assets/24300129/374e032f-4353-4a39-b853-649844b99885">
<img width="1792" alt="image" src="https://github.com/craftcms/cms/assets/24300129/bc61d57c-1ac5-4dcb-86e8-ebe231b3f45a">
<img width="1792" alt="image" src="https://github.com/craftcms/cms/assets/24300129/080020b1-253b-4877-9ec0-22bd93eb6475">
<img width="1789" alt="image" src="https://github.com/craftcms/cms/assets/24300129/e0d28960-5eb9-45a4-84fd-47753d22eaf7">
<img width="1792" alt="image" src="https://github.com/craftcms/cms/assets/24300129/8bb574e2-cdd5-4b73-b168-ed64ce936204">
<img width="1792" alt="image" src="https://github.com/craftcms/cms/assets/24300129/13b6ba1a-3742-44d4-88c4-1a0d4ecd910e">
<img width="1792" alt="image" src="https://github.com/craftcms/cms/assets/24300129/5990cd06-45dc-497e-9141-08bb06f2721b">
<img width="1792" alt="image" src="https://github.com/craftcms/cms/assets/24300129/1258197b-9c8d-43ac-94ee-25a7c3ae7ef3">
<img width="1792" alt="image" src="https://github.com/craftcms/cms/assets/24300129/97905531-6e15-47c8-a5e6-5967aecf1c84">
<img width="1792" alt="image" src="https://github.com/craftcms/cms/assets/24300129/1dd036b6-3e2d-449d-a876-beae93ded554">
<img width="1792" alt="image" src="https://github.com/craftcms/cms/assets/24300129/ecff16a8-22b6-4a6c-aaa6-65f14c7392b9">
<img width="1792" alt="image" src="https://github.com/craftcms/cms/assets/24300129/938c218d-ef67-47ee-af33-a0cdc8df315c">
<img width="1129" alt="image" src="https://github.com/craftcms/cms/assets/24300129/d21d4f42-1ce7-4936-89b0-dec40eb6379f">

**Comment by cherrykoda** [comment]
This is sort of a duplicate of this thread - https://github.com/craftcms/cms/discussions/9062

**Comment by kellyannmcnamara01** [comment]
@brandonkelly, I think there was confusion from @cherrykoda. I am not looking for a work around. I am looking for a way to convert thousands of entries that use a field that is no long supported and need a way to transfer that content to a different solution without losing days worth of work. It appears this was prematurely closed due.

To be clear:
A field we use in the site is a plugin (link text field) and it is getting discontinued. We need a way to smoothly transfer thousands of page's of content to a different solution. Similar to how Redactor and CKeditor were able to run a command line task.

**Reply by brandonkelly** [reply]
It looks to me like this is fundamentally the same request as #9062 (which is at its core a feature request, not a proposed workaround).

**Comment by brandonkelly** [comment, marked as answer]
Craft 5.3.0-beta.1 is out with a new Link field type 🎉 (#15251).

We’re targeting August 6 for the GA release. If you’d like to test ahead of time, change your `craftcms/cms` requirement in composer.json to:

```json
"craftcms/cms": "^5.3.0-beta.1",
```

and then run `composer update`.
